kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/ModemManagerDbusWrapper.java

```diff
@@ -250,119 +248,69 @@ private List<Properties> getBearersPropertiesFromPaths(List<DBusPath> bearerPath
 
     }
 
-    protected void resetHandlerEnable(String deviceId, Optional<String> modemManagerDbusPath, int delayMinutes,
-            String networkManagerDbusPath) throws DBusException {
-        if (!modemManagerDbusPath.isPresent()) {
-            logger.warn("Cannot retrieve modem device for {}. Skipping modem reset monitor setup.", deviceId);
+    protected void modemTaskHandler(String deviceId, Device device, NetworkProperties properties) throws DBusException {
+        int resetDelayMinutes = properties.get(Integer.class, "net.interface.%s.config.resetTimeout", deviceId);
+        boolean autoconnect = properties.get(Boolean.class, "net.interface.%s.config.persist", deviceId);
+        if (isModemTaskAlreadyActivated(deviceId, resetDelayMinutes, autoconnect)) {
             return;
         }
+        modemTaskHandlerDisable(deviceId);
```

Comment:
Is this correct? You're checking whether the task was activated, if not you deactivate it?

Reply:
The task is activated if the handler is present and the reset or autoconnect task is enabled. In this case, there is nothing to do. 
Otherwise, that is if the handler is not present or it is present but the reset and autoconnect task is not enabled, we need to recreate it.